objects/src/transaction/proven_tx.rs

```diff
@@ -22,6 +24,7 @@ pub struct ProvenTransaction {
     consumed_notes: Vec<ConsumedNoteInfo>,
     created_notes: Vec<NoteEnvelope>,
     tx_script_root: Option<Digest>,
+    program_hash: Digest,
```

Comment:
In theory, we should be able to compute this from consumed notes and tx script root, right?

Reply:
Yes, that's correct. If we remove the program hash then we will incur some additional computational costs of computing the program hash. If we keep it then we have an additional 256 bits of size per proven tx.  What do you think is preferable? I'll add a TODO inline if we want to compute it.

Reply:
Thinking about this deeper I think we will have to compute the hash otherwise there is a trust assumption that the provided program hash is associated with the provided tx_script and notes.

Reply:
implemented in #180